libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -317,13 +331,45 @@ bool TransactionManager::checkBlockTransactions(DagBlock const &blk) {
     for (auto const &tx_hash : blk.getTrxs()) {
       if (!transactions_pool_.contains(tx_hash) && !nonfinalized_transactions_in_dag_.contains(tx_hash) &&
           !db_->transactionFinalized(tx_hash)) {
-        LOG(log_er_) << "Block " << blk.getHash() << " has missing transaction " << tx_hash;
-        return false;
+        if (low_nonce_transactions_.count(tx_hash)) {
+          LOG(log_dg_) << "Block " << blk.getHash() << " has low nonce transaction " << tx_hash;
+        } else if (insufficient_balance_transactions_.count(tx_hash)) {
+          LOG(log_dg_) << "Block " << blk.getHash() << " has insufficient balance transaction " << tx_hash;
+        } else {
+          LOG(log_er_) << "Block " << blk.getHash() << " has missing transaction " << tx_hash;
+          return false;
+        }
```

Comment:
same here. The rule is to eliminate nested if conditions as much as possible because it makes the code easier to read:

for (...) {
  if (transactions_pool_.contains(tx_hash) || nonfinalized_transactions_in_dag_.contains(tx_hash) ||
          db_->transactionFinalized(tx_hash)) {
     continue;
  }

  // rest of the code
}

Reply:
also this could be inside transactions_pool_.contains